### PR TITLE
chore: Changed managed API to use common code.

### PR DESCRIPTION
**Pull request by abhidixi11**
Use helper functions in iot_mqtt_serialize.c.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/src/iot_mqtt_helper.c [hunk 1]
```diff
@@ -80,8 +83,96 @@
...
+
+/*-----------------------------------------------------------*/
+
+static uint8_t * _encodeUserNameAndMetrics( uint8_t * pDestination,
+                                            const IotMqttConnectInfo_t * pConnectInfo,
+                                            bool * pEncodedUserName )
+{
+    uint8_t * pBuffer = pDestination;
+
+    #if AWS_IOT_MQTT_ENABLE_METRICS == 1
```
Comment (nateglims) [inline libraries/standard/mqtt/src/iot_mqtt_helper.c]: Do we test with both metrics enabled and disabled?

File: libraries/standard/mqtt/src/private/iot_mqtt_helper.h [hunk 2]
```diff
@@ -71,7 +71,17 @@
  * @param[in] x The unsigned int to check.
  * @param[in] position Which bit to check.
  */
-#define UINT8_CHECK_BIT( x, position )    ( ( ( x ) & ( 0x01U << ( position ) ) ) == ( 0x01U << ( position ) ) )
+#define UINT8_CHECK_BIT( x, position )    ( ( ( x ) &( 0x01U << ( position ) ) ) == ( 0x01U << ( position ) ) )
```
Comment (nateglims) [inline libraries/standard/mqtt/src/private/iot_mqtt_helper.h]: I think this formatting is slightly off.

File: libraries/standard/mqtt/src/private/iot_mqtt_helper.h [hunk 3]
```diff
@@ -278,6 +288,6 @@ IotMqttError_t _IotMqtt_ProcessIncomingPublishFlags( uint8_t publishFlags,
  * @return Pointer to the end of the encoded string, which will be identical to
  * `pDestination` if nothing was encoded.
  */
- uint8_t * _IotMqtt_EncodeUserName( uint8_t * pDestination,
-                                  const IotMqttConnectInfo_t * pConnectInfo );
-#endif
\ No newline at end of file
+uint8_t * _IotMqtt_EncodeUserName( uint8_t * pDestination,
+                                   const IotMqttConnectInfo_t * pConnectInfo );
+#endif /* ifndef IOT_MQTT_SERIALIZE_INTERNAL_H_ */
```
Comment (nateglims) [inline libraries/standard/mqtt/src/private/iot_mqtt_helper.h]: It looks like it was this way before the change, but is the header guard intentionally different from the file name?

File: libraries/standard/mqtt/src/iot_mqtt_helper.c (hunk 1 above)
Comment (abhidixi11) [reply libraries/standard/mqtt/src/iot_mqtt_helper.c]: It depends on the configuration. I think we test with enabled , but my local setup it was disabled. So didn't this problem.

File: libraries/standard/mqtt/src/private/iot_mqtt_helper.h (hunk 3 above)
Comment (abhidixi11) [reply libraries/standard/mqtt/src/private/iot_mqtt_helper.h]: Good catch. It is cut and paste error. Will fix it.

File: libraries/standard/mqtt/src/private/iot_mqtt_helper.h (hunk 2 above)
Comment (abhidixi11) [reply libraries/standard/mqtt/src/private/iot_mqtt_helper.h]: I'll run uncrustify again.

**Pull request merged**